### PR TITLE
Dataset metadata publishing #2

### PR DESCRIPTION
**Pull request by MLumme**
Grouping of tasks https://www.pivotaltracker.com/story/show/185927876, https://www.pivotaltracker.com/story/show/185927742, https://www.pivotaltracker.com/story/show/185927868 and https://www.pivotaltracker.com/story/show/185927934. 

Testable at https://185927876.dev.laji.fi/theme/dataset-metadata/.

### CODE REVIEW

**Review by aorin — CHANGES_REQUESTED** [review]

File: projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata-browser/dataset-metadata-browser.component.scss [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  max-height: calc(100vh - 50px);
+}
+
+.selector-container {
+  flex: 1;
+  display: flex;
+  flex-direction: column;
+}
+
+.selector-empty-height {
```
[the remark this answers is not in the document]
Comment (MLumme) [reply projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata-browser/dataset-metadata-browser.component.scss:17]: Not used, removed.

File: projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata.component.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+.container {
+  display: flex;
+  width: 100%;
+  height: 100%;
+}
+
+.sidebar {
+  min-height: calc(100vh - 50px - 384px);
```
[the remark this answers is not in the document]
Comment (MLumme) [reply projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata.component.scss:19]: Originally I added that as it seemed that the sidebar wanished if no collection was selected and it was minimized, but seems it was some odd interaction with the other styles, as it doesn't happen with these changes.

File: projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata.component.ts [hunk 3]
```diff
@@ -46,6 +49,14 @@ export class DatasetMetadataComponent implements OnInit, AfterViewInit {
     if (this.isMobile) {
       this.showBrowser = false;
     }
+
+    this.paramSub$ = this.route.params.subscribe(params => {
+      this.updateCollection(params.collectionId);
```
[the remark this answers is not in the document]
Comment (MLumme) [reply projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata.component.ts:54]: Done.

File: projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata.component.ts [hunk 4]
```diff
@@ -13,17 +14,19 @@ const mobileBreakpoint = 768;
   styleUrls: ['./dataset-metadata.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class DatasetMetadataComponent implements OnInit, AfterViewInit {
+export class DatasetMetadataComponent implements OnInit, OnDestroy, AfterViewInit {
   collectionId: string;
   _collectionId: string;
```
[the remark this answers is not in the document]
Comment (MLumme) [reply projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata.component.ts:19]: And done.

File: projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata.component.ts [hunk 5]
```diff
@@ -60,32 +71,33 @@ export class DatasetMetadataComponent implements OnInit, AfterViewInit {
     if (this.collectionId && this.collectionId !== this._collectionId) {
       this.collection$ = this.collectionService.getById$(this.collectionId, 'multi');
       this.collectionCounts$ = this.collectionService.getCollectionSpecimenCounts$(this.collectionId);
-
+      this._collectionId = this.collectionId;
       this.cd.markForCheck();
     }
   }
 
   changeUrl(collectionId) {
```
[the remark this answers is not in the document]
Comment (MLumme) [reply projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata.component.ts:79]: Should be done.

File: projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata.component.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (aorin) [reply projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata.component.scss]: I was able to broke the sidebar by resizing the screen with firefox so I added a commit to fix it. Can you check that it's fine?

[the remark this answers is not in the document]
Comment (MLumme) [reply projects/laji/src/app/shared-modules/dataset-metadata/dataset-metadata.component.scss]: Will do, and I'm not surprised something broke, it's a pretty finiky thing with so many nested levels from all the components.

**Review by aorin — APPROVED** [review]
Thanks, looks good!

**Pull request merged**